### PR TITLE
Fi-323: Fix FHIR version checking

### PR DESCRIPTION
**Pull request by Jammjammjamm**
Currently, when trying to run a a module with no version specified (such as the SMART module), tests will be skipped because the FHIR versions don't match. This branch updates the version checking to properly support modules with no version requirement.

I added Minitest as an explicit dependency so that we aren't just getting it through one of our other dependencies. I also discovered that MiniTest supports spec-style tests, so I converted the TestingInstance tests. I think this style makes it a lot easier to write small tests with a clear intention and am looking for feedback from the team.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-323
- [x] Internal ticket links to this PR
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [x] Code diff has been reviewed for extraneous/missing code
- [x] Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name: Rob Scanlon
- [X] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [X] The tests appropriately test the new code, including edge cases
- [X] You have tried to break the code

### CODE REVIEW

File: test/unit/test_instance_test.rb [hunk 1]
```diff
@@ -2,51 +2,76 @@
...
-    Inferno::Models::ServerCapabilities.create(
-      testing_instance_id: @instance.id,
-      capabilities: conformance.as_json
-    )
-
-    assert !@instance.conformance_supported?(:Patient, [:read])
-    assert !@instance.conformance_supported?(:Patient, [:read, :search, :history])
-    assert !@instance.conformance_supported?(:Patient)
-    assert @instance.conformance_supported?(:Observation, [:read])
+      assert !@instance.conformance_supported?(:Patient)
```
[the remark this answers is not in the document]
Comment (Jammjammjamm) [reply test/unit/test_instance_test.rb]: I also don't have a strong opinion.

**Comment by Jammjammjamm** [comment]
It would be nice to have one way to determine whether FHIR versions match, but I think that might be out of scope for this ticket. Maybe I should give the method a more specific name to make it clear that it's only for checking whether the instance and sequence match.

File: Gemfile [hunk 2]
```diff
@@ -11,6 +11,7 @@ gem 'dm-sqlite-adapter'
 gem 'fhir_client'
 gem 'json-jwt'
 gem 'kramdown'
+gem 'minitest'
```
Comment (arscan) [inline Gemfile]: Should this be in the `group :test`?

**Comment by arscan** [comment]
Also, this test approach does seem more explicit, though we might be able to accomplish the same thing using the old test format, but just writing them better :)

File: Gemfile (hunk 2 above)
Comment (Jammjammjamm) [reply Gemfile]: I'm pretty sure this is where we get `assert`, so it's used outside of tests.

**Comment by Jammjammjamm** [comment]
This may just be my ignorance, but with the other style tests, I don't know how to do scoped setup. I often find it useful to do things like:
```
describe SomeClass do
  before { global_setup}

  describe '#some_method' do
    before { setup_just_for_testing_this_method }
  ...
```

File: Gemfile (hunk 2 above)
Comment (radamson) [reply Gemfile]: The `assert` in the sequences comes from the `Inferno::Assertions` module if that's what you're referring to.

https://github.com/onc-healthit/inferno/blob/dbb146a7bef896790c6e4695dde6520406740039/lib/app/utils/assertions.rb#L6-L10

Comment (Jammjammjamm) [reply Gemfile]: Well, that seems obvious in retrospect.

**Comment by arscan** [comment]
> This may just be my ignorance, but with the other style tests, I don't know how to do scoped setup.

I suppose you would just call a setup method at the beginning of the test?  Yeah it isn't very clear from a semantic perspective, but you could address by how you name the method?  I am not very opinionated on this though.
